infrastructure/templates/public-api/components/functionApp.bicep

```diff
@@ -183,58 +190,94 @@ resource slot2FileShare 'Microsoft.Storage/storageAccounts/fileServices/shares@2
   ]
 }
 
+var commonSiteProperties = {
+ enabled: true
+ httpsOnly: true
+ serverFarmId: appServicePlan.id
+
+ // This property integrates the Function App into a VNet given the supplied subnet id.
+ virtualNetworkSubnetId: subnetId
+
+ clientAffinityEnabled: true
+ reserved: reserved
+ siteConfig: {
+   alwaysOn: alwaysOn ?? null
+   preWarmedInstanceCount: preWarmedInstanceCount ?? null
+   netFrameworkVersion: '8.0'
+   linuxFxVersion: appServicePlanOS == 'Linux' ? 'DOTNET-ISOLATED|8.0' : null
+   keyVaultReferenceIdentity: keyVaultReferenceIdentity
+ }
+ keyVaultReferenceIdentity: keyVaultReferenceIdentity
+ publicNetworkAccess: publicNetworkAccessEnabled ? 'Enabled' : 'Disabled'
+}
+
+// Create the main production deploy slot.
 resource functionApp 'Microsoft.Web/sites@2023-01-01' = {
   name: fullFunctionAppName
   location: location
   kind: 'functionapp'
   identity: identity
-  properties: {
-    enabled: true
-    httpsOnly: true
-    serverFarmId: appServicePlan.id
-
-    // This property integrates the Function App into a VNet given the supplied subnet id.
-    virtualNetworkSubnetId: subnetId
-
-    clientAffinityEnabled: true
-    reserved: reserved
-    siteConfig: {
-      alwaysOn: alwaysOn ?? null
-      preWarmedInstanceCount: preWarmedInstanceCount ?? null
-      netFrameworkVersion: '8.0'
-      linuxFxVersion: appServicePlanOS == 'Linux' ? 'DOTNET-ISOLATED|8.0' : null
-      keyVaultReferenceIdentity: keyVaultReferenceIdentity
-    }
-    keyVaultReferenceIdentity: keyVaultReferenceIdentity
-    publicNetworkAccess: publicNetworkAccessEnabled ? 'Enabled' : 'Disabled'
-  }
+  properties: commonSiteProperties
   tags: tagValues
 }
 
-// TODO EES-5128 - reuse the majority of properties from the production slot configuration above
-// (e.g. enables, httpsOnly etc) rather than duplicating manually here.
+// Create the staging deploy slot.
 resource stagingSlot 'Microsoft.Web/sites/slots@2023-01-01' = {
   name: 'staging'
   parent: functionApp
   location: location
   identity: identity
-  properties: {
-    enabled: true
-    httpsOnly: true
-    serverFarmId: appServicePlan.id
-
-    // This property integrates the Function App slot into a VNet given the supplied subnet id.
-    virtualNetworkSubnetId: subnetId
+  properties: commonSiteProperties
+  tags: tagValues
+}
 
-    clientAffinityEnabled: true
-    reserved: reserved
-    siteConfig: {
-      keyVaultReferenceIdentity: keyVaultReferenceIdentity
+var authSettingsV2Properties = entraIdAuthentication == null ? {} : {
+  globalValidation: {
+    requireAuthentication: true
+    unauthenticatedClientAction: 'Return401'
+  }
```

Comment:
Hi yes.  That'll be because I was trying all sorts of different solutions during Friday, and I ended up leaving the auth disabled right at the end.  The health checks has been working successfully prior to this though.  Once I can get a successful deploy out again, I'll show that it's working OK with the auth (although I'm doubting myself now)!

I think that it works by virtue of the pings coming from the host machine itself against the containers running within it, and it is therefore able to bypass the auth layer.